src/main/java/de/wirvsvirus/hack/rest/AvatarController.java

```diff
@@ -28,72 +30,95 @@
+  }
 
-    @GetMapping(value = STOCK_AVATAR_ENDPOINT + "/{stockAvatar}")
-    public ResponseEntity<Resource> getStockAvatar(@NotNull @PathVariable("stockAvatar") final StockAvatar stockAvatar) {
-        CacheControl cacheConfiguration = CacheControl
-                .maxAge(1, TimeUnit.DAYS)
-                .cachePublic();
-        return copyResourceToResponse(avatarService.getStockAvatarResource(stockAvatar), cacheConfiguration);
-    }
+  @GetMapping(value = STOCK_AVATAR_ENDPOINT + "/{stockAvatar}")
+  public ResponseEntity<Resource> getStockAvatar(
+      @NotNull @PathVariable("stockAvatar") final StockAvatar stockAvatar) throws Throwable {
+    CacheControl cacheConfiguration = CacheControl
+        .maxAge(1, TimeUnit.DAYS)
+        .cachePublic();
+    return copyResourceToResponse(avatarService.getStockAvatarResource(stockAvatar),
+        cacheConfiguration);
+  }
 
-    private ResponseEntity<Resource> copyResourceToResponse(final ClassPathResource avatarResource, final CacheControl cacheConfiguration) {
-        final MediaType responseContentType;
-        if (avatarResource.getFilename().endsWith(".png")) {
-            responseContentType = MediaType.IMAGE_PNG;
-        } else if (avatarResource.getFilename().endsWith(".jpg")) {
-            responseContentType = MediaType.IMAGE_JPEG;
-        } else {
-            throw new IllegalStateException(String.format("Unknown image file extension: %s", avatarResource.getFilename()));
-        }
+  @GetMapping(value = STOCK_AVATAR_SVG_ENDPOINT + "/{stockAvatar}")
+  public ResponseEntity<Resource> getStockAvatarSvg(
+      @NotNull @PathVariable("stockAvatar") final StockAvatar stockAvatar) throws Throwable {
+    CacheControl cacheConfiguration = CacheControl
+        .maxAge(1, TimeUnit.DAYS)
+        .cachePublic();
+    return copyResourceToResponse(avatarService.getStockAvatarSvgResource(stockAvatar),
+        cacheConfiguration);
+  }
 
-        return ResponseEntity.ok()
-                .contentType(responseContentType)
-                .eTag(getResourceHash(avatarResource))
-                .cacheControl(cacheConfiguration)
-                .body(avatarResource);
-    }
+  private ResponseEntity<Resource> copyResourceToResponse(final ClassPathResource avatarResource,
+      final CacheControl cacheConfiguration) throws Throwable {
+    final MediaType responseContentType = Optional.ofNullable(avatarResource.getFilename())
+        .map(filename -> {
+          if (filename.endsWith(".png")) {
+            return MediaType.IMAGE_PNG;
+          } else if (filename.endsWith(".jpg")) {
+            return MediaType.IMAGE_JPEG;
+          } else if (filename.endsWith(".svg")) {
+            return new MediaType("image", "svg+xml");
+          } else {
+            throw new IllegalStateException(
+                String.format("Unknown image file extension: %s", filename));
+          }
+        }).orElseThrow(() -> {
```

Comment:
nit: avatarResource.getFilename() != null würde ich schon ganz oben in der Methode prüfen